### PR TITLE
Autofill Blob Headers in Platform SDK

### PR DESCRIPTION
**Pull request by nihalbhatnagar**

### CODE REVIEW

File: packages/client/src/ontology/loadFullObjectMetadata.ts
Comment (nihalbhatnagar) [inline packages/client/src/ontology/loadFullObjectMetadata.ts:1]: Had to make this change do to a change over the wire. Underlying API URL's remained consistent

Comment (ericanderson) [reply packages/client/src/ontology/loadFullObjectMetadata.ts:1]: Makes sense, they are still evolving some of these.

File: packages/client/src/util/toDataValue.ts [hunk 1]
```diff
@@ -55,10 +55,6 @@ export async function toDataValue(
       {
         filename: value.name,
       },
-      {
-        "Content-Length": value.size.toString(),
-        "Content-Type": value.type,
```
Comment (ericanderson) [inline packages/client/src/util/toDataValue.ts:60]: So obviously in this case, we just want to use whatever the blob's content-type is. After all, we're asking it.

However, the foundry platform sdk is used by customers too and they may at some point end up with a blob that has a content type they don't want. Lets say they get an upload for a json file but they need the content type in the system to be something specific? 

Therefore, I would like to raise the question: Should we make `Content-Type` optional and fill it with the blob value if not provided?

Comment (nihalbhatnagar) [reply packages/client/src/util/toDataValue.ts:60]: I see the use case, and I probably agree with the solution. The alternative would be to just enforce the users attach a Blob with the correct type attached, but that's probably more overhead than we should pass to the consumers and also requires them to make a copy. I don't think there's a scenario in which the Blob's actual attached content type is transmitted through the wire or used in the generated SDK's, so this should be fine

**Review by ericanderson — APPROVED** [review]

File: packages/foundry.core/src/_components.ts [hunk 2]
```diff
@@ -233,8 +233,7 @@ export type TimeUnit =
   | "DAYS"
   | "WEEKS"
   | "MONTHS"
-  | "YEARS"
-  | "QUARTERS";
+  | "YEARS";
```
Comment (ssanjay1) [inline packages/foundry.core/src/_components.ts:236]: Could you track as a FLUP to see if we need to remove support for `QUARTERS` in our internal time unit types?

File: packages/platform-sdk-generator/src/generateResource2.ts [hunk 3]
```diff
@@ -66,21 +68,38 @@ async function generateMethods(resource: Resource, model: Model) {
...
+        && shouldFillBlobHeaders
+        ? blobHeaders.autofill
+        : ""
+    }return $foundryPlatformFetch($ctx, _${methodName}, ${
+      shouldFillBlobHeaders ? blobHeaders.returnArgs : "...args"
+    }); }
 
     `;
   }
+
```
Comment (ssanjay1) [inline packages/platform-sdk-generator/src/generateResource2.ts:102]: nit:  can we remove this newline

**Review by ssanjay1 — APPROVED** [review]

**Pull request merged**